DON'T MERGE. IT WILL BE RECREATED. android-app

iZIVer opened the pull request:
* user-friendly android app

jhollowayj commented on examples/android-app/cubicsdk/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">CubicSdk</string>
```
`Cubic Example Client`, etc. might be a better name.

iZIVer commented on examples/android-app/app/src/main/java/com/cubic/example/toast/ToastManager.java:
```diff
@@ -0,0 +1,115 @@
+        mToast = Toast.makeText(mContext, message, duration);
+        mToast.show();
+        return mToast;
+    }
```
В планах было описать интерфейс для разных способов отображения сообщений, например  snackbar. Ок сделаю попроще

iZIVer replied:
The plans were to describe the interface for different ways of displaying messages, for example snackbar. Ok i'll make it simpler

iZIVer replied:
Updated interface for abstract message manager

iZIVer commented on examples/android-app/app/src/main/java/com/cubic/example/toast/ToastManager.java:
```diff
@@ -0,0 +1,115 @@
+            mToast = Toast.makeText(mContext, message, duration);
+        } else {
+            mToast = Toast.makeText(mContext, message, duration);
+        }
```
Of course, drawable was used for custom ui, I removed it

iZIVer commented on examples/android-app/app/src/main/res/raw/recognition.json:
```diff
@@ -0,0 +1 @@
+{"v":"5.3.1","fr":120,"ip":0,"op":241,"w":600,"h":600,"nm":"Comp 2","ddd":0,"assets":[],"layers":[{"ddd":0,"ind":1,"ty":4,"nm":"Shape Layer 20","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[300,300,0],"ix":2},"a":{"a":0,"k":[0,0,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6}},"ao":0,"hasMask":true,"masksProperties":[{"inv":false,"mode":"f","pt":{"a":1,"k":[{" ...
```
At first I wanted to add animation to record the microphone so that the application was interactive, but then I abandoned this idea)

iZIVer commented on examples/android-app/app/src/main/res/layout/dialog_connection.xml:
```diff
@@ -0,0 +1,21 @@
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <LinearLayout
```
sure)

iZIVer commented on examples/android-app/app/src/main/res/layout/dialog_network.xml:
```diff
@@ -0,0 +1,21 @@
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <LinearLayout
```
Already corrected

iZIVer commented on examples/android-app/app/src/main/res/layout/dialog_settings.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
If you need to make a custom layout dialog, then you need it. In the meantime, I removed it

iZIVer commented on examples/android-app/app/build.gradle:
```diff
@@ -0,0 +1,43 @@
+    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+
+    implementation 'com.airbnb.android:lottie:3.0.0-beta2'
```
Decided not to add lottie yet

iZIVer commented on examples/android-app/cubicsdk/build.gradle:
```diff
@@ -0,0 +1,93 @@
+        versionName "1.0"
+
+        buildConfigField "String", "HOST", '"demo-cubic.cobaltspeech.com:2727"'
+        buildConfigField "Boolean", "SECURE_CONNECTION", 'true'
```
Yes, I decided to specify the parameters for connecting to the server, but in the future there was an idea to use RemoteConfig from Firebase to change the connection parameters. And I wanted to make a fully standalone application for the user. You can place these settings in a separate settings window.

iZIVer commented on examples/android-app/cubicsdk/src/main/proto/cubic.proto:
```diff
@@ -0,0 +1,359 @@
+// Copyright (2019) Cobalt Speech and Language Inc.
```
I agree, I will correct in the near future

iZIVer replied:
Added a script in gradle for loading proto files from the repository

iZIVer commented on examples/android-app/cubicsdk/src/main/proto/google/api/annotations.proto:
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015, Google Inc.
```
I agree

iZIVer replied:
1. Now I have made it so that the connection dialog does not close. I have thoughts on how to connect the connection indication with the talk button, I will try to implement it.

2. The fall of the application corrected

3. Did you mean the settings for the recording model? Yes, I think to display the selected model as a settings icon.

jhollowayj replied:
1. ok.
2. great!
3. see my other response.

iZIVer replied:
Added a script in gradle for loading proto files from the repository

jhollowayj commented:
The code changes seem reasonable.  I'll do some manual testing in the morning.  I think it will pass though.

iZIVer commented:
In the last commit:
- Added a spinner to select models at the top of the main screen.
- In the application settings made changes to the host and the security connection

iZIVer commented:
In the last commit:
* fixed bug with displaying connection error
* when a connection is lost, the list of moles is cleared

iZIVer commented:
In the future, it is better to make progress interactive so that it does not block the interface as a dialogue. I think the best solution would be to display a snackbar at the top or combine with the button talk

* removed the lock of the connection dialog
* added timeout check

jhollowayj commented:
Alright.  I think this looks good.  There's a lot of code that is unrelated (directly) to using the the gRPC/sdk-cubic stuff, so I think we need to add in an Android specific README.md that explains which files are most applicable.  Maybe we could include a few screenshots of the application as well, explaining where to change the server, how to start transcribing, etc.

We should also have a readme on what they would need to do to include the SDK in their application.  That needs to be very clear and straight forward.

This is also replacing the existing android application, so I would suggest you remove the `examples/android` folder, and rename your folder from `examples/android-app` to `examples/android-app`.

In terms of cleaning things up, I would suggest we squash all of these files into one or two commits, and then drop the old android app folder in another commit, then move these files over to `examples/android`.  I think this PR should be reduced to 4-5 commits total.
